demo/api/src/pages/pages-entity-info.service.ts

```diff
@@ -0,0 +1,18 @@
+import { PageTreeService } from "@comet/cms-api";
+import { EntityInfoServiceInterface } from "@comet/cms-api/lib/dam/files/decorators/entity-info.decorator";
+import { Injectable } from "@nestjs/common";
+import { Page } from "@src/pages/entities/page.entity";
+
+@Injectable()
+export class PagesEntityInfoService implements EntityInfoServiceInterface<Page> {
+    constructor(private readonly pageTreeService: PageTreeService) {}
```

Comment:
The PageTreeReadApi is pretty optimized for reading multiple page tree nodes in one request. It does that by caching (per request)

When you create each time a new instance (createReadApi) you always reset the caches. Changing that is very simple, just inject `PageTreeReadApiService` directly.

And the have most advantage of the cache you can preload the pagetree for a scope:
`readApi.preloadNodes(pageTreeNode.scope)`
- can be called multiple times (will preload only once)
- only helpful if you will work with many nodes
- nodePath is already working with all parent nodes

Reply:
we might also consider moving this service into the library as it will be used exactly like this for all projects

Reply:
Alright, I wasn't aware of the `PageTreeReadApiService`. Thanks for the info. I changed that. And I also moved the service to the cms-api Library.

Reply:
Now after moving the `PagesEntityInfoService` to the library, suddenly the injected `PageTreeReadApiService` is always undefined. I worked around this by going back to injecting `PageTreeService`.

Do you have an idea why it doesn't work? There is no Nest error that it can't inject the dependency. It's simply undefined.

Reply:
I'll keep using `pageTreeService.createReadApi` for simplicitys sake